File: app/src/main/java/com/aibfive/sample/ui/tencentmap/TencentMapActivity.java
package com.aibfive.sample.ui.tencentmap;

import android.content.Context;
import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;

import com.aibfive.sample.R;

/**
 * 腾讯地图
 */
public class TencentMapActivity extends AppCompatActivity {

    public static void start(Context context) {
        Intent starter = new Intent(context, TencentMapActivity.class);
        context.startActivity(starter);
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_tencent_map);
    }

    /**
     * 发送定位
     * @param view
     */
    public void onSendLocationClick(View view){
        SendLocationActivity.start(this);
    }

    /**
     * 搜索地址
     * @param view
     */
    public void onSearchAddressClick(View view){
        SearchAddressActivity.Companion.start(this);
    }
}

File: app/src/main/java/com/aibfive/sample/ui/tencentmap/TencentMapActivity.kt
package com.aibfive.sample.ui.tencentmap

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.view.View
import com.aibfive.basetools.ui.BaseActivity
import com.aibfive.sample.R
import com.aibfive.sample.databinding.ActivityTencentMapBinding

class TencentMapActivity : BaseActivity<ActivityTencentMapBinding>(), View.OnClickListener {
    
    companion object{

        @JvmStatic
        fun start(context: Context) {
            val starter = Intent(context, TencentMapActivity::class.java)
            context.startActivity(starter)
        }

    }

    override fun initView() {
        super.initView()
        binding.tvSendLocation.setOnClickListener(this)
        binding.tvSearchAddress.setOnClickListener(this)
    }


    override fun onClick(v: View?) {
        if (v != null) {
            when(v.id){
                R.id.tv_send_location -> SendLocationActivity.start(this)
                R.id.tv_search_address -> SearchAddressActivity.start(this)
            }
        }
    }
}